java/arcs/core/data/proto/SchemaProtoDecoder.kt

```diff
@@ -24,21 +24,22 @@ fun SchemaProto.decodeNames() = getNamesList().map { SchemaName(it) }.toSet()
 fun SchemaProto.decodeFields(): SchemaFields {
     val singletons = mutableMapOf<FieldName, FieldType>()
     val collections = mutableMapOf<FieldName, FieldType>()
-    for ((name, type) in getFieldsMap()) {
-        when (type.getDataCase()) {
+    for ((name, type) in fieldsMap) {
+        when (type.dataCase) {
             TypeProto.DataCase.PRIMITIVE -> singletons[name] = type.decodeAsFieldType()
-            // TODO: TypeProto.DataCase.COLLECTIONS
-            // TODO: TypeProto.DataCase.REFERENCE
+            TypeProto.DataCase.REFERENCE -> singletons[name] = type.decodeAsFieldType()
+            TypeProto.DataCase.TUPLE -> singletons[name] = type.decodeAsFieldType()
```

Comment:
Since everything updates `singletons` map, how about?
```suggestion
             TypeProto.DataCase.REFERENCE, 
             TypeProto.DataCase.PRIMITIVE, 
             TypeProto.DataCase.TUPLE -> singletons[name] = type.decodeAsFieldType()
```

Reply:
Nice, I like this!